### ISSUE TITLE
error server.lua

### ISSUE THREAD

**Issue by MrAlmagro** [Elipse458/el_business#9]
Hello @Elipse458
I have recently downloaded your script, and i have just opened it on my server and this appears:
SCRIPT ERROR: @el_business/server.lua:251: attempt to index a nil value (global ‘ESX’)

This is written on that lines
`
function reloadPlayersData()
local xPlayers = ESX.GetPlayers()
if #xPlayers<1 then return end
TriggerClientEvent(“el_business:syncServer”,-1,dataCache)
end
`

**Comment by Elipse458** [comment]
try restarting the resource lol

**Comment by Elipse458** [comment]
if that doesn't work you might be using an event scrambler or something that changes the `esx:getSharedObject` event

**Comment by MrAlmagro** [comment]
Then, what should i writte?

**Comment by MrAlmagro** [comment]
I have alreasy fixed it. But how can i do for it to make income even if the person is disconected??
Ty if you answer

**Comment by Elipse458** [comment]
change the sql query on [line 197](https://github.com/Elipse458/el_business/blob/master/server.lua#L197)

**Comment by MrAlmagro** [comment]
What should i change??


`function addMoney(identifier,money,business)
    if not identifier then return end
    local xPlayer = ESX.GetPlayerFromIdentifier(identifier)
    local money = math.floor(tonumber(money))
    if xPlayer then
        print("[^1"..GetCurrentResourceName().."^7] Adding money to "..GetPlayerName(xPlayer.source).." - "..identifier.." ("..tostring(money).."$)")
        TriggerClientEvent('esx:showNotification', xPlayer.source, _L("received_money"):format(tostring(money),business))
        xPlayer.addAccountMoney("bank", math.floor(money))
    else
        print("[^1"..GetCurrentResourceName().."^7] An error occured while adding money to "..identifier.." ("..tostring(money).."$). The player is offline. Forcing adding money")
        MySQL.Sync.execute('UPDATE `users` SET `bank` = `bank` + @bank WHERE `identifier` = @identifier',{['@bank'] = money, ['@identifier'] = identifier})
    end
end`

**Comment by Elipse458** [comment]
are you using latest esx?

**Comment by MrAlmagro** [comment]
Yep

**Comment by Elipse458** [comment]
well then, like i said, change the sql query to be compatible with it

**Comment by MrAlmagro** [comment]
Yes, now It is compatible with my server, and It is working. But i want to
know, if people could get money even when offline.

If is is already configured like that, pls tell It to me

**Comment by Elipse458** [comment]
i already told you twice what you need to do

**Comment by MrAlmagro** [comment]
You told me the line i have to change, but not what to put!!
Pls help me

**Comment by Elipse458** [comment]
idk what you need to put there, i don't use latest esx, you'll need to figure that out by yourself

**Comment by MrAlmagro** [comment]
Which one should i use then?
So that i can configure that?

**Comment by Elipse458** [comment]
v1.1.0

**Comment by dimoni666** [comment]
> Hello @Elipse458
> I have recently downloaded your script, and i have just opened it on my server and this appears:
> SCRIPT ERROR: @el_business/server.lua:251: attempt to index a nil value (global ‘ESX’)
> 
> This is written on that lines
> `
> function reloadPlayersData()
> 
> local xPlayers = ESX.GetPlayers()
> 
> if #xPlayers<1 then return end
> 
> TriggerClientEvent(“el_business:syncServer”,-1,dataCache)
> 
> end
> 
> `
> 
> Now it has also appeared in line 262

Can u tell how u fix it pls??? Thank you :) @MrAlmagro
